Learning_test_automation/Spotkanie6_Zadania/CurrencyConversion_vBeta/Option/CurrencyPresentation.cs

```diff
@@ -41,27 +65,37 @@ private void PrintAvailableCurrencies()
             Console.WriteLine("- Czech crown: 3.\n");
         }
 
-        public void PrintAvailableBuyExchangeRates()
+        public void PrintAvailableListExchangeRatesValue()
         {
+            string caller = new StackTrace().GetFrame(1).GetMethod().Name;
+
             Console.WriteLine("\nHere are the values of currencies that we have prepared for you: ");
-            foreach (CurrencyDefault currency in Enum.GetValues(typeof(CurrencyDefault)))
+
+            foreach (var currency in currencies)
             {
-                if (currency == CurrencyDefault.Dollar) Console.WriteLine($"- {dollar.NameCurrency}: {dollar.BoughtCurrency} pln,");
-                else if (currency == CurrencyDefault.Euro) Console.WriteLine($"- {euro.NameCurrency}: {euro.BoughtCurrency} pln,");
-                else if (currency == CurrencyDefault.Czech_crown) Console.WriteLine($"- {czechCrown.NameCurrency}: {czechCrown.BoughtCurrency} pln.");
+                name = currency.NameCurrency.ToString();
+
+                if (caller == "Buying") Console.WriteLine($"- {name}: {currency.BoughtCurrency} pln,");
+                else Console.WriteLine($"- {name}: {currency.SoldCurrency} pln,");
             }
+
             Console.WriteLine();
         }
-        public void PrintAvailableSellExchangeRates()
+
+        public void PrintAvailableBuyAndSellExchangeRates()
         {
-            Console.WriteLine("\nHere are the values of currencies that we have prepared for you: ");
-            foreach (CurrencyDefault currency in Enum.GetValues(typeof(CurrencyDefault)))
-            {
-                if (currency == CurrencyDefault.Dollar) Console.WriteLine($"- {dollar.NameCurrency}: {dollar.SoldCurrency} pln,");
-                else if (currency == CurrencyDefault.Euro) Console.WriteLine($"- {euro.NameCurrency}: {euro.SoldCurrency} pln,");
-                else if (currency == CurrencyDefault.Czech_crown) Console.WriteLine($"- {czechCrown.NameCurrency}: {czechCrown.SoldCurrency} pln.");
-            }
-            Console.WriteLine();
+            Transaction transaction = new Transaction();
+
+            Console.Write($"I understand my dear user, you have chosen to print the value of each available currency.\nEnter the value to be converted: ");
+
+            decimal amount = option.GetAmount();
```

Comment:
Po co? :p

Reply:
Bo wydawało mi się że jak coś jest częściej niż raz to lepiej żeby miał zmienną 🗡️